### PR TITLE
feat: adds initial support for CIRA APF messages and handlers

### PR DESCRIPTION
**Pull request by rsdmike**

### CODE REVIEW

File: pkg/apf/types.go [hunk 1]
```diff
@@ -245,6 +260,20 @@ type APF_PROTOCOL_VERSION_MESSAGE struct {
...
 	MessageType byte
 }
+
+/**
+ * holds the user authentication request failure response.
+ * @MessageType - APF_USERAUTH_FAILURE
+ * @AuthenticationsThatCanContinueLength - length of the methods string
+ * @AuthenticationsThatCanContinue - methods that can continue (e.g., "password")
+ * @PartialSuccess - 0 for no partial success.
+ *.*/
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/types.go]: Fixed - changed `.*/` to `*/` in the comment.

File: pkg/apf/processor.go [hunk 2]
```diff
@@ -323,12 +756,47 @@ func ProcessProtocolVersion(data []byte) APF_PROTOCOL_VERSION_MESSAGE {
...
+
+	// Rearrange according to GUID format (little-endian for first 3 groups)
+	guid := hexStr[6:8] + hexStr[4:6] + hexStr[2:4] + hexStr[0:2] + "-" +
+		hexStr[10:12] + hexStr[8:10] + "-" +
+		hexStr[14:16] + hexStr[12:14] + "-" +
+		hexStr[16:20] + "-" +
+		hexStr[20:]
+
+	return guid
+}
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/processor.go]: Added test coverage - TestBytesToHex and TestHexToGUID now provide comprehensive tests for these helper functions including edge cases.

File: pkg/apf/processor.go [hunk 3]
```diff
@@ -353,19 +821,41 @@ func ServiceAccept(serviceName string) APF_SERVICE_ACCEPT_MESSAGE {
...
+	log.Debug("sending APF_KEEPALIVE_OPTIONS_REQUEST_MESSAGE")
+
+	message := APF_KEEPALIVE_OPTIONS_REQUEST_MESSAGE{
+		MessageType:     APF_KEEPALIVE_OPTIONS_REQUEST,
+		IntervalSeconds: keepAliveTime,
+		TimeoutSeconds:  timeout,
+	}
+
+	return message
+}
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/processor.go]: Added test coverage - TestKeepAliveOptionsRequest now tests this function.

File: pkg/apf/processor.go [hunk 4]
```diff
@@ -88,6 +286,238 @@ func Process(data []byte, session *Session) bytes.Buffer {
...
+		// failureMessage := &APF_USERAUTH_FAILURE_MESSAGE{
+		// 	MessageType:                          APF_USERAUTH_FAILURE,
+		// 	AuthenticationsThatCanContinueLength: uint32(len("password")),
+		// 	AuthenticationsThatCanContinue:       []byte("password"),
+		// 	PartialSuccess:                       0,
+		// }
+		// return failureMessage
+		return nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/processor.go]: ProcessUserAuthRequest is kept for backward compatibility - existing code that calls this function directly (outside of the Processor flow) will continue to work. It can be deprecated in a future release once consumers migrate to the new Processor-based API.

File: pkg/apf/handler.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+type DefaultHandler struct{}
+
+// OnProtocolVersion accepts all protocol version messages.
+func (h DefaultHandler) OnProtocolVersion(info ProtocolVersionInfo) error {
+	return nil
+}
+
+// OnAuthRequest does not authenticate by default.
+func (h DefaultHandler) OnAuthRequest(request AuthRequest) AuthResponse {
+	return AuthResponse{Authenticated: false}
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/handler.go]: This is intentional - DefaultHandler.OnAuthRequest returns false because callers MUST provide a custom Handler implementation to perform actual authentication. The old ProcessUserAuthRequest function that always returned success is kept for backward compatibility but the new Processor-based API requires explicit authentication handling.

File: pkg/apf/processor.go [hunk 6]
```diff
@@ -323,12 +756,49 @@ func ProcessProtocolVersion(data []byte) APF_PROTOCOL_VERSION_MESSAGE {
...
+	guid := hexStr[6:8] + hexStr[4:6] + hexStr[2:4] + hexStr[0:2] + "-" +
+		hexStr[10:12] + hexStr[8:10] + "-" +
+		hexStr[14:16] + hexStr[12:14] + "-" +
+		hexStr[16:20] + "-" +
+		hexStr[20:]
+
+	return guid
+}
+
 // Send the AFP service accept message to the MEI.
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/processor.go]: Fixed - changed 'AFP' to 'APF' in the comment.

File: pkg/apf/processor.go [hunk 7]
```diff
@@ -16,17 +16,190 @@ import (
...
+	case APF_KEEPALIVE_REQUEST:
+		log.Debug("received APF_KEEPALIVE_REQUEST")
+
+		dataToSend = ProcessKeepAliveRequest(data, session)
+	case APF_KEEPALIVE_REPLY:
+		log.Debug("received APF_KEEPALIVE_REPLY")
+
+		// 	dataToSend = ProcessKeepAliveReply(data, session)
+	case APF_KEEPALIVE_OPTIONS_REPLY:
+		log.Debug("received APF_KEEPALIVE_OPTIONS_REQUEST")
```
[the remark this answers is not in the document]
Comment (rsdmike) [reply pkg/apf/processor.go]: Fixed - changed log message to 'APF_KEEPALIVE_OPTIONS_REPLY'.

**Review by madhavilosetty-intel — APPROVED** [review]
